editor/script_create_dialog.cpp

```diff
@@ -892,22 +892,41 @@ ScriptLanguage::ScriptTemplate ScriptCreateDialog::_parse_template(const ScriptL
 					script_template.description = line.substr(12, -1).strip_edges();
 				}
 				if (line.begins_with("space-indent")) {
-					String indent_value = line.substr(17, -1).strip_edges();
+					String indent_value = line.substr(13, -1).strip_edges();
 					if (indent_value.is_valid_int()) {
 						int indent_size = indent_value.to_int();
 						if (indent_size >= 0) {
-							space_indent = String(" ").repeat(indent_size);
+							space_indent_size = indent_size;
 						}
 					} else {
-						WARN_PRINT(vformat("Template meta-use_space_indent need to be a valid integer value. Found %s.", indent_value));
+						WARN_PRINT(vformat("Template meta-space-indent need to be a valid integer value. Found %s.", indent_value));
 					}
 				}
 			} else {
-				// Store script
-				if (space_indent != "") {
-					line = line.replace(space_indent, "_TS_");
+				// Replace indentation.
+				int i = 0;
+				int space_count = 0;
+				while (i < line.length() && (line[i] == ' ' || line[i] == '\t')) {
+					if (line[i] == '\t') {
+						line = line.left(i) + "_TS_" + line.substr(i + 1);
+						space_count = 0;
+						i += 4;
+						continue;
+					}
+					space_count++;
+
+					if (space_count != space_indent_size) {
+						++i;
+						continue;
+					}
+
+					line = line.left(i - (space_indent_size - 1)) + "_TS_" + line.substr(i + 1);
+
+					space_count = 0;
+					i += -(space_indent_size - 1) + 4;
 				}
-				script_template.content += line.replace("\t", "_TS_") + "\n";
+
+				script_template.content += line + "\n";
```

Comment:
Can this be simplified to something like:

```c++
int space_count = 0;
for (int i = 0; i < line.length(); i++) {
    if (line[i] == '\t') {
        script_template.content += "_TS_";
    } else if (line[i] == ' ') {
        space_count++;
        if (space_count == space_indent_size) {
            script_template.content += "_TS_";
            space_count = 0;
        }
    } else {
        script_template.content += line.substr(i) + "\n";
        break;
    }
}
```

Reply:
Not completely matching but will see about adding a variation of this simplification, thank you!

Reply:
Which all in all is why I chose to use the method from `CodeEdit` with modifications to keep it more simple while still ensuring the behavior, but will see if I can do something similar while still maintaining that aspect of the original behavior (the part ahead of the first non-space character)

Reply:
Will test the following when I'm back and able to test the code:
```cpp
int space_count = 0;
for (int i = 0; i < line.length(); i++) {
    if (line[i] == '\t') {
        if (space_count) {
            script_template.content += String(" ").repeat(space_count);
            space_count = 0;
        }
        script_template.content += "_TS_";
    } else if (line[i] == ' ') {
        space_count++;
        if (space_count == space_indent_size) {
            script_template.content += "_TS_";
            space_count = 0;
        }
    } else {
        if (space_count) {
            script_template.content += String(" ").repeat(space_count);
            space_count = 0;
        }
        script_template.content += line.substr(i) + "\n";
        break;
    }
}
```

This will deal with the spaces case